### DISCUSSION TITLE
Change the Expression of a (existing) field Parameter Table via script

### DISCUSSION THREAD

**Discussion by NaHolla** [TabularEditor/TabularEditor3#1403] [🙋♀️TE3 Q&A] [answered]
I have a table which is defind as a field Parameter (which I've customized)

TableName = "TOP-Kategorie"

Now, I'd like to update the expression

var targetTable = Model.Tables["TOP-Kategorie"];

var MyNewExpression = ReadFile(@"U:...\TabularMeasures\TOP-Kategorie.csv"); // this works

targetTable.Expression = MyNewExpression ; //not working

![image](https://github.com/user-attachments/assets/e2de5a32-db2c-4556-a8ed-56e340c8f5e0)
how do I change the exression? Thanks!

**Comment by otykier** [comment]
Are you connected to Power BI Desktop while trying to run this script? If so, could you try to enable the **Allow unsupported modeling operations** option under **Tools > Preferences > Power BI Desktop**:

![image](https://github.com/user-attachments/assets/e8ede4ac-7bd5-4e17-89de-dd10405a5aca)

**Reply by NaHolla** [reply]
Hi,
I´m connected to PowerBI. I'v enebled the setting, but it still comes to an error, (transladed from german) :
TabularEditor.TomWrapper.Table contains no definition for Expression and no MethodExpression could be found

**Reply by NaHolla** [reply]
error code CS1061

**Reply by otykier** [reply, marked as answer]
oh I see - you didn't mention that you got an error message in your original post - so I just assumed the script ran without error but didn't change anything.

In any case, it's an easy fix. You just need to explicitly cast the table to type `CalculatedTable` before attempting to access the `Expression` property. Like so:

```csharp
var targetTable = Model.Tables["TOP-Kategorie"];
var MyNewExpression = ReadFile(@"U:...\TabularMeasures\TOP-Kategorie.csv"); // this works

// Explicit cast to CalculatedTable should let you assign the Expression property:
(targetTable as CalculatedTable).Expression = MyNewExpression ;
```

**Reply by NaHolla** [reply]
Thanks! now it is working!
